### PR TITLE
SDDM theme with Autologin (My first PR)

### PR DESCRIPTION
**Pull request by mashrukk**
# Pull Request

## SDDM Astronaut theme with Autologin
This script installs sddm-astronaut-theme from last stream and provides an option to enable autologin with sddm. Users can choose which desktop session they want to autologin on startup. After selecting desktop session, it will do a sddm restart and autologin to that desktop.

## Testing
If you want to test it first, do the following.
```
git clone --branch sddm https://github.com/mashrukk/linutil
cd linutil && cargo run
```

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: tabs/applications-setup/sddm-theme.sh [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            i=$((i + 1))
+        done
+    done
+
+    # Prompt user to choose a session
+    while true; do
+        printf "Enter the number of the session you'd like to autologin: "
+        read -r choice
+        if [ "$choice" -ge 1 ] && [ "$choice" -lt "$i" ]; then
+            session_file="/usr/share/$session_type/$(sed -n "${choice}p" <<< "$(printf "%s\n" /usr/share/$session_type/*.desktop)")"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:84]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        fi
+    done
+
+    # Update SDDM configuration
+    $ESCALATION_TOOL sed -i "1i[Autologin]\nUser = $USERNAME\nSession = $actual_session\n" /etc/sddm.conf
+    printf "Autologin configuration complete.\n"
+}
+
+checkEnv
+checkEscalationTool
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:98]:
```
checkEnv
checkEscalationTool
```

These need to be at the bottom of the script.

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:98]: These are at the bottom of the script. It looks like this.

checkEnv
checkEscalationTool
install_sddm
install_theme
configure_autologin

Comment (nnyyxxxx) [reply tabs/applications-setup/sddm-theme.sh:98]: They are not at the bottom of the script @mashrukk

Comment (nnyyxxxx) [reply tabs/applications-setup/sddm-theme.sh:98]: image

Comment (adamperkowski) [reply tabs/applications-setup/sddm-theme.sh:98]: You should probably move theese:

https://github.com/ChrisTitusTech/linutil/blob/3be8dbcc083d3e1d4071f18e611b1fc531d8f331/tabs/applications-setup/sddm-theme.sh#L101-L105

https://github.com/ChrisTitusTech/linutil/blob/3be8dbcc083d3e1d4071f18e611b1fc531d8f331/tabs/applications-setup/sddm-theme.sh#L109-L115

to the functions.

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:98]: I saw in other scripts in applications-setup, checkEnv and checkEscalationTool is at first then we do other functions. Right? @nnyyxxxx

Comment (nnyyxxxx) [reply tabs/applications-setup/sddm-theme.sh:98]: > I saw in other scripts in applications-setup, checkEnv and checkEscalationTool is at first then we do other functions. Right? @nnyyxxxx

checkEnv is always first, The rest are usually ordered in the placement of the functions.

Ex: Randomfunction, Randomfunction4, Randomfunction1, Randomfunction9

checkEnv 
Randomfunction
Randomfunction4
Randomfunction1
Randomfunction9

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:98]: @nnyyxxxx What about checkEscalationTool. Does it come after checkEnv?

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:101]: @nnyyxxxx Check the full script now. Any issues?

Comment (nnyyxxxx) [reply tabs/applications-setup/sddm-theme.sh:101]: Many @mashrukk

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:103]: Oh crap, wait

Comment (mashrukk) [reply tabs/applications-setup/sddm-theme.sh:102]: @nnyyxxxx I reverted the changes. I have an exam tomorrow. Can you please fix the issues and share the updated version? That'd be great.

**Comment by nnyyxxxx** [comment]
@mashrukk You'll have to revert these 3 commits, too many breaking changes in them. After that move these to the bottom of the script

```
checkEnv
checkEscalationTool
```

image

**Comment by mashrukk** [comment]
@nnyyxxxx I reverted the changes. I have an exam tomorrow. Can you please fix the issues and share the updated version? That'd be great.

File: tabs/applications-setup/sddm-theme.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            printf "%d) %s (%s)\n" "$i" "$name" "$type"
+            sessions[$i]="$session_file"
+            session_names[$i]="$name ($type)"
+            i=`expr $i + 1`
+        done
+    done
+
+    # Prompt user to choose a session
+    while true; do
+        printf "Enter the number of the session you'd like to autologin: "
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:84]: Every printf without a color code should be changed to ``echo``

File: tabs/applications-setup/sddm-theme.sh [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    # Find the corresponding .desktop file and Update SDDM configuration
+    actual_session=`basename "$session_file" .desktop`
+
+    $ESCALATION_TOOL sed -i '1i[Autologin]\nUser = '$USERNAME'\nSession = '$actual_session'\n' /etc/sddm.conf
+    echo "Autologin configuration complete."
+}
+
+checkEnv
+checkEscalationTool
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:102]: These still need to be moved to the bottom of the script.

**Comment by adamperkowski** [comment]
@mashrukk I could try to fix this and open a PR on your branch if you want to.

**Comment by mashrukk** [comment]
@nnyyxxxx Check. Sorry I'm bothering you with so many commits

**Comment by adamperkowski** [comment]
nyx is afk

File: tabs/applications-setup/sddm-theme.sh [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            ;;
+        dnf)
+            $ESCALATION_TOOL dnf update
+            $ESCALATION_TOOL dnf install -y sddm qt6-qtsvg
+            ;;
+        pacman)
+            $ESCALATION_TOOL pacman -S --needed --noconfirm sddm qt6-svg
+            ;;
+        *)
+            printf "${RED}Unsupported package manager. Please install SDDM manually.${RC}\n"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:26]: Every ``printf`` with a color code needs to be changed to ``printf "%b\n"``

File: tabs/applications-setup/sddm-theme.sh (hunk 4 above)
Comment (nnyyxxxx) [reply tabs/applications-setup/sddm-theme.sh:102]: > These still need to be moved to the bottom of the script.

@mashrukk

File: tabs/applications-setup/sddm-theme.sh [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            session_type="xsessions"
+        else
+            session_type="wayland-sessions"
+        fi
+
+        for session_file in /usr/share/$session_type/*.desktop; do
+            [ -e "$session_file" ] || continue
+            name=`grep -i "^Name=" "$session_file" | cut -d= -f2`
+            type=`echo $session_type | sed 's/s$//'`  # Remove trailing 's'
+            echo "%d) %s (%s)\n" "$i" "$name" "$type"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:75]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+            session_names[$i]="$name ($type)"
+            i=`expr $i + 1`
+        done
+    done
+
+    # Prompt user to choose a session
+    while true; do
+        echo "Enter the number of the session you'd like to autologin: "
+        read choice
+        if [ -n "${sessions[$choice]}" ]; then
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:86]: This is not posix compliant.

File: tabs/applications-setup/sddm-theme.sh [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+            session_file="${sessions[$choice]}"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/sddm-theme.sh:87]: This is not posix compliant.

**Comment by mashrukk** [comment]
@nnyyxxxx Can you please make a PR on my sddm branch?

**Comment by mashrukk** [comment]
> @mashrukk I could try to fix this and open a PR on your branch if you want to. Drft this PR for now.

Sure @adamperkowski

**Pull request closed without merging**